src/main/groovy/org/pih/warehouse/core/identification/BlankIdentifierResolver.groovy

```diff
@@ -55,9 +55,7 @@ trait BlankIdentifierResolver<T extends GormEntity> {
                 log.info("Assigning identifier ${identifier} to ${entityType} with id: ${entity.id}")
                 setIdentifierOnEntity(identifier, entity)
 
-                // TODO: investigate why this merge with flush and no validation is required. Can we simplify this to
-                //       a regular save operation?
-                if (!entity.merge(flush: true, validate: false)) {
+                if (!entity.save()) {
```

Comment:
from what I've gathered, the merge was required because the assign identifier job was wrapping all the generate id calls with the same `Product.withNewSession`, even though only one of them was actually saving to product. This meant that the entities might not have been in the session, and so we were forced to work with detached objects via merge. Now that we're leaving the transactions to the individual services themselves, which do the work to fetch the entities (and thus load them into the session) this is no longer required.